test/controllers/products_controller_test.rb

```diff
@@ -0,0 +1,228 @@
+require "test_helper"
+
+describe ProductsController do
+  describe 'guest user' do
+    describe 'index' do
+      it 'succeeds with multiple products for a guest user' do
```

Comment:
You need to include negative cases for the things a guest user / the wrong user can't do.